compiler/noirc_driver/src/lib.rs

```diff
@@ -131,6 +131,56 @@ pub struct CompileOptions {
     pub skip_underconstrained_check: bool,
 }
 
+/// Similar to CompileOptions but with some extra fields specific for `check_crate`.
+#[derive(Clone, Debug, Default)]
+pub struct CheckOptions {
```

Comment:
If we want this to be a superset, let's just have a CompileOptions field internally

Reply:
Sounds good. I thought the same thing while having lunch, I pushed this change right before lunch 😀